packages/happy-dom/src/query-selector/SelectorItem.ts

```diff
@@ -5,10 +5,13 @@ import Element from '../nodes/element/Element';
 const ATTRIBUTE_REGEXP =
 	/\[([a-zA-Z0-9-_]+)\]|\[([a-zA-Z0-9-_]+)([~|^$*]{0,1})[ ]*=[ ]*["']{0,1}([^"']+)["']{0,1}\]/g;
 const ATTRIBUTE_NAME_REGEXP = /[^a-zA-Z0-9-_$]/;
-const PSUEDO_REGEXP = /:([a-zA-Z-]+)\(([0-9n+-]+|odd|even)\)|:not\(([^)]+)\)|:([a-zA-Z-]+)/g;
-const CLASS_REGEXP = /\.([a-zA-Z0-9-_$]+)/g;
+const PSUEDO_REGEXP =
+	/(?<!\\):([a-zA-Z-]+)\(([0-9n+-]+|odd|even)\)|(?<!\\):not\(([^)]+)\)|(?<!\\):([a-zA-Z-]+)/g;
+const CLASS_REGEXP = /\.(([a-zA-Z0-9-_$]|\\:)+)/g;
 const TAG_NAME_REGEXP = /^[a-zA-Z0-9-]+/;
 const ID_REGEXP = /#[A-Za-z][-A-Za-z0-9_]*/g;
+const CSS_ESCAPE_REGEXP = /(?<!\\):/g;
```

Comment:
this escape should be more than just for `:` but don't know if I can just have it be any arbitrary character or a set of characters that can be escaped like `[:#&...]`